### ISSUE TITLE
Excluding a nested object field causes subsequent fields to be null

### ISSUE THREAD

**Issue by juyttenh** [elastic/elasticsearch-hadoop#811]
When a field of a nested object is excluded, all values of the (subsequent) fields of the nested object are ignored resulting in null values.
### Steps to reproduce

Create index and insert data:

```
curl -XPUT 'localhost:9200/companies/?pretty=1' -d '{
  "mappings": {
    "company": {
      "properties": {
        "name": { "type": "string"},
        "description": {"type": "string"},
        "employees": {
          "type": "nested",
          "properties": {
            "name": {"type": "string"},
            "age": {"type": "integer"},
            "salary": {"type": "long"}
          }
        }
      }
    }
  }
}'

curl -XPUT 'localhost:9200/companies/company/1' -d '{
  "name": "mycompany",
  "employees": [
    { "name": "anne",
      "age": 30, 
      "salary": 6},
    { "name": "bob",
      "age": 55,
      "salary": 100},
    { "name": "charlie",
      "age": 22,
      "salary": 15
    }],
  "description": "a cool tech startup"
}'
```

This shows correct results:

```
scala> val options = Map("pushdown" -> "true", "es.field.read.as.array.include" -> "employees")
scala> val esDf = sqlContext.read.format("org.elasticsearch.spark.sql").options(options).load("companies/company")
scala> esDf.show
```

This however shows incorrect results: it gives employees fields with null values, when only the field name should be excluded: 

```
scala> val options = Map("pushdown" -> "true", "es.field.read.as.array.include" -> "employees", "es.read.field.exclude" -> "employees.name")
scala> val esDf = sqlContext.read.format("org.elasticsearch.spark.sql").options(options).load("companies/company")
scala> esDf.show
```

The problem seems to be caused by org.elasticsearch.hadoop.serialization.ScrollReader that skips (the remainders of) an entire block instead when it `shouldSkip` just the excluded field.
To demonstrate: if I replace [skipCurrentBlock](https://github.com/elastic/elasticsearch-hadoop/blob/2.x/mr/src/main/java/org/elasticsearch/hadoop/serialization/ScrollReader.java#L798) with the following statements

```
parser.nextToken();
parser.skipChildren();
parser.nextToken();
```

... at least for this case I get the expected results
### Version Info

OS: Mac OSX 10.11.5
JVM : Scala version 2.10.5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_20-ea)
Hadoop/Spark : Spark-1.6.2
ES-Hadoop : org.elasticsearch:elasticsearch-spark_2.10:2.3.3
ES : elasticsearch-2.3.3

**Comment by ranlei** [comment]
Maybe you can written this: `val options = Map("pushdown" -> "true", "es.field.read.as.array.include" -> "employees")`

**Comment by jbaiera** [comment]
This seems to be [fixed in 5.0.0](https://github.com/elastic/elasticsearch-hadoop/commit/dd1d6036fa9ee21d9a3082226a152d40cce9ede7#diff-98b0b8f1a4399ff0a3d596156d752e1cR790). I'm going to close this out.
